PR: https://github.com/odoo/odoo/pull/215789

From: edc6562db9a8eff0bf5b92bbda17d36af30927f3
From: alan-odoo
Date: 2025-09-06 21:40:47

Breaking data model changes scores: del:9 + add:0, change matches:
-    _inherit = 'res.users'
-    resume_line_ids = fields.One2many(related='employee_id.resume_line_ids', readonly=False)
-    employee_skill_ids = fields.One2many(related='employee_id.employee_skill_ids')
-    current_employee_skill_ids = fields.One2many('hr.employee.skill', related="employee_id.current_employee_skill_ids", readonly=False)
-    certification_ids = fields.One2many('hr.employee.skill', related="employee_id.certification_ids", readonly=False)

Total Changes: 37

[IMP] *: update the res.users form views

This commit modifies the fields on the preferences / profile form, which
allows users to read or modify their own data depending on their
permissions, and on the user form for the administrators. The aim of this
change is to display more relevant information to users.

The hr.res_users_view_form_simple_modif view has been removed and merged in
the mail_bot_hr.res_users_view_form_simple_modif view since it must not be
used anymore for the preferences form as this one must not have a different
display when the hr module is installed.

As some fields are removed from the forms and are not longer used, they are
also removed from the res.users model.

Enterprise PR: https://github.com/odoo/enterprise/pull/88518
Upgrade PR: https://github.com/odoo/upgrade/pull/8207

Task-4775990

Part-of: odoo/odoo#215789
Related: odoo/enterprise#88518
Related: odoo/upgrade#8207
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills/models/__init__.py
+++ b/addons/hr_skills/models/__init__.py
@@ -11,5 +11,4 @@ from . import hr_employee_skill
 from . import hr_job_skill
 from . import hr_skill_level
 from . import hr_skill_type
-from . import res_users
 from . import resource_resource

--- a/addons/hr_skills/models/res_users.py
+++ b/None
@@ -1,36 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-from odoo import fields, models
-
-
-class ResUsers(models.Model):
-    _inherit = 'res.users'
-
-    resume_line_ids = fields.One2many(related='employee_id.resume_line_ids', readonly=False)
-    employee_skill_ids = fields.One2many(related='employee_id.employee_skill_ids')
-    current_employee_skill_ids = fields.One2many('hr.employee.skill', related="employee_id.current_employee_skill_ids", readonly=False)
-    certification_ids = fields.One2many('hr.employee.skill', related="employee_id.certification_ids", readonly=False)
-
-    @property
-    def SELF_READABLE_FIELDS(self):
-        return super().SELF_READABLE_FIELDS + [
-            'resume_line_ids',
-            'employee_skill_ids',
-            'current_employee_skill_ids',
-            'certification_ids',
-        ]
-
-    @property
-    def SELF_WRITEABLE_FIELDS(self):
-        return super().SELF_WRITEABLE_FIELDS + [
-            'resume_line_ids',
-        ]
-
-    def write(self, vals):
-        if 'current_employee_skill_ids' in vals or 'certification_ids' in vals or 'employee_skill_ids' in vals:
-            vals['employee_skill_ids'] = vals.pop('current_employee_skill_ids', []) + vals.pop('certification_ids', []) + vals.get('employee_skill_ids', [])
-
-        # Must be called directly on employee_id to prevent SET values in vals from causing unintended behavior
-        if 'employee_skill_ids' in vals:
-            self.employee_id.write({'employee_skill_ids': vals.pop("employee_skill_ids")})
-        res = super().write(vals)
-        return res
